src/Development/Ninja/Lexer.hs

```diff
@@ -137,22 +141,30 @@ lexBind c_x | (c,x) <- list0 c_x = case c of
     '\0' -> []
     _ -> lexxBind LexBind c_x
 
+lexBuild :: Str0 -> [Lexeme]
 lexBuild x
     | (outputs,x) <- lexxExprs True x
     , (rule,x) <- span0 isVarDot $ dropSpace x
     , (deps,x) <- lexxExprs False $ dropSpace x
     = LexBuild outputs rule deps : lexerLoop x
 
+lexDefault :: Str0 -> [Lexeme]
 lexDefault x
     | (files,x) <- lexxExprs False x
     = LexDefault files : lexerLoop x
 
-lexRule x = lexxName LexRule x
-lexPool x = lexxName LexPool x
-lexInclude x = lexxFile LexInclude x
-lexSubninja x = lexxFile LexSubninja x
-lexDefine x = lexxBind LexDefine x
-
+lexRule :: Str0 -> [Lexeme]
```

Comment:
All should share one decl, e.g. `lexRule, lexPool, lexInclude :: Str0 -> [Lexeme]`